src/Microsoft.DotNet.Build.Tasks/Targets/publishtest.targets

```diff
@@ -83,7 +83,7 @@
       OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)"
       Retries="$(CopyRetryCount)"
       RetryDelayMilliseconds="$(CopyRetryDelayMilliseconds)"
-      UseHardlinksIfPossible="true">
+      UseHardlinksIfPossible="$(CreateHardLinksForCopyFilesToOutputDirectoryIfPossible)">
```

Comment:
Ah, was about to ask how to detect host OS in msbuild, but that would be good enough for the common case of iterating on Windows.